### PR TITLE
Extract favorite local storage code to own file

### LINKED ISSUE #91: Extract all the favorite local storage code to its own file
ATM it is scattered among several files (```harbour-berlin-vegan.qml```, ```IconToolBar.qml```, etc.).

### PR DESCRIPTION
**Pull request by jmastr**
Do not talk to the SQLite database directly.

Sources:
* http://doc.qt.io/qt-5/qtquick-localstorage-localstorage-database-js.html

Closes #91

### CODE REVIEW

File: qml/components-generic/Database.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/**
+ *
+ *  This file is part of the Berlin-Vegan guide (SailfishOS app version),
+ *  Copyright 2015-2016 (c) by micu <micuintus.de> (micuintus@gmx.de).
```
Comment (micuintus) [inline qml/components-generic/Database.js]: Oh, we need to change our headers! :)

Comment (jmastr) [reply qml/components-generic/Database.js]: Therefore we have #150 ;)

File: qml/harbour-berlin-vegan.qml [hunk 2]
```diff
@@ -56,29 +52,17 @@ ApplicationWindow
...
-    
-    function openDataBase() {
-        db = LocalStorage.openDatabaseSync("BerlinVeganDB", "0.1", "Berlin Vegan SQL!", 1000000);
-        db.transaction(function(tx) {
-            tx.executeSql("CREATE TABLE IF NOT EXISTS BerlinVegan(favorite_id TEXT)");
-            favorite_ids = tx.executeSql("SELECT favorite_id FROM BerlinVegan");
-        });
-    }
-
-    function applyFavoritesFromDataBase()
```
Comment (micuintus) [inline qml/harbour-berlin-vegan.qml]: Just wondering: Is there a specific reason why inlined this back into the ```favoritesHook()```

**Comment by micuintus** [comment]
Nicce. Paying debts sucks, but once it is done, it fells so good. Thanks!

File: qml/harbour-berlin-vegan.qml (hunk 2 above)
Comment (jmastr) [reply qml/harbour-berlin-vegan.qml]: You mean the `favourite_ids` part? I could not access `gJsonVenueModel` directly and it just seemed wrong the pass it into a `BVApp.Database.dbXYZ()` function.

**Comment by jmastr** [comment]
> Nicce. Paying debts sucks, but once it is done, it fells so good. Thanks!

Agreed!

File: qml/harbour-berlin-vegan.qml (hunk 2 above)
Comment (micuintus) [reply qml/harbour-berlin-vegan.qml]: Yes, agreed.
But we could leave it here? But that's a minor…

Comment (jmastr) [reply qml/harbour-berlin-vegan.qml]: Then please merge as is.

**Pull request merged**